src/client/components/map/MapToken.tsx

```diff
@@ -262,6 +267,42 @@ function MapTokenInner({
       )}
     </Container>
   );
+  const notifications = useServerState((state) => state.logEntries);
+  const [lastRolled, setLastRolled] = useState<RRLogEntryDiceRoll | null>();
+  useEffect(() => {
+    const list = entries(notifications);
+
+    function checkEntrySuitable(
+      entry: RRLogEntry
+    ): entry is RRLogEntryDiceRoll {
+      return (
+        entry.type === "diceRoll" &&
+        entry.payload.characterIds !== null &&
+        entry.payload.characterIds.includes(character.id) &&
+        Date.now() - entry.timestamp < ENTRY_DURATION
+      );
+    }
+
+    let i = list.length - 1;
+    let entry: RRLogEntry | null = list[i] ?? null;
+    while (entry && !checkEntrySuitable(entry)) {
+      if (Date.now() - entry.timestamp > ENTRY_DURATION) {
```

Comment:
I wonder whether the duplicate check of the timestamp could be avoided if we used a loop like this:

```js
let selectedEntry = null;
for (let i = list.length - 1, entry = list[i]; !selectedEntry && entry && entry not too old; entry = list[--i]) {
  if (entry is diceRoll and for my character) {
    selectedEntry = entry;
  }
}
```

But maybe that makes it less readable :D

Reply:
I think it could only get more readable :D I think it's looking better now